toolchain/check/eval_inst.cpp

```diff
@@ -4,16 +4,41 @@
 
 #include "toolchain/check/eval_inst.h"
 
+#include <variant>
+
 #include "toolchain/check/action.h"
 #include "toolchain/check/facet_type.h"
 #include "toolchain/check/generic.h"
+#include "toolchain/check/impl_lookup.h"
 #include "toolchain/check/import_ref.h"
 #include "toolchain/check/type.h"
 #include "toolchain/check/type_completion.h"
+#include "toolchain/sem_ir/ids.h"
 #include "toolchain/sem_ir/typed_insts.h"
 
 namespace Carbon::Check {
 
+// When calling from eval to various Check functions, we need the actual LocId.
+// This allows us to unwrap the SemIRLoc to do so.
+//
+// TODO: We could do this unwrap at the entrance to eval, and work with LocId
+// everywhere in eval, instead of deferring it to points where we exit eval back
+// into Check.
```

Comment:
From discussion, I assume you're expecting more things to need similar logic. I think this is approach is good right now since it stops bigger changes from blocking this PR, but for the TODO, I think we probably don't want to keep unwrapping here. Whether that means `LocId` everywhere or `SemIRLoc` everywhere, I'm not sure, but I've brought it up [on discord](https://discord.com/channels/655572317891461132/655578254970716160/1353755830058745959).

Maybe:

```suggestion
// TODO: Decide whether to refactor calls everywhere to accept `SemIRLoc`, or
// fold `SemIRLoc` into `LocId`. Either way, we would like eval to call other
// code without unwrapping `SemIRLoc`.
```

Reply:
Done